### PR TITLE
Allow `string` to mode cross

### PR DESCRIPTION
**Pull request by liam923**
Modify the definition of `string` in predef to allow crossing along linearity, contention, and portability.

### CODE REVIEW

**Review by goldfirere — APPROVED** [review]

File: ocaml/testsuite/tests/typing-modes/val_modalities.ml [hunk 1]
```diff
@@ -277,7 +277,7 @@ module Close_over_value :
 
 module Close_over_value_monadic = struct
   module M = struct
-    let r @ uncontended = "hello"
+    let r @ uncontended = fun x -> x
```
Comment (riaqn) [inline ocaml/testsuite/tests/typing-modes/val_modalities.ml:280]: Might be better to use `int ref`  (because arrows will cross contention)

**Review by riaqn — COMMENTED** [review]
I reviewed `typing-modes` and `typing-unique`.

File: ocaml/testsuite/tests/typing-modes/val_modalities.ml (hunk 1 above)
Comment (riaqn) [reply ocaml/testsuite/tests/typing-modes/val_modalities.ml:280]: Unfortunately the currently mode-checking wrt mutable fields is too strict. Can you add a `(* CR mode-crossing: this should use mutable record *)` and I will fix in later PRs?

Comment (liam923) [reply ocaml/testsuite/tests/typing-modes/val_modalities.ml:281]: Done

**Pull request merged**